src/bilder/images/docker_baseline_ami/deploy.py

```diff
@@ -0,0 +1,143 @@
+
+VERSIONS = {
+    "consul": os.environ.get("CONSUL_VERSION", CONSUL_VERSION),
+    "vault": os.environ.get("VAULT_VERSION", VAULT_VERSION),
+    "consul_template": os.environ.get(
+        "CONSUL_TEMPLATE_VERSION", CONSUL_TEMPLATE_VERSION
+    ),
+}
+TEMPLATES_DIRECTORY = Path(__file__).parent.joinpath("templates")
+VECTOR_INSTALL_NAME = "vector-log"
+
+# Set up configuration objects
+set_env_secrets(Path("consul/consul.env"))
+
+consul_configuration = {Path("00-default.json"): ConsulConfig()}
+vector_config = VectorConfig(is_proxy=False)
+
+vault_config = VaultAgentConfig(
+    cache=VaultAgentCache(use_auto_auth_token="force"),
+    listener=[
+        VaultListener(
+            tcp=VaultTCPListener(
+                address=f"127.0.0.1:{VAULT_HTTP_PORT}", tls_disable=True
+            )
+        )
+    ],
+    vault=VaultConnectionConfig(
+        address=f"https://active.vault.service.consul:{VAULT_HTTP_PORT}",
+        tls_skip_verify=True,
+    ),
+    auto_auth=VaultAutoAuthConfig(
+        method=VaultAutoAuthMethod(
+            type="aws",
+            mount_path="auth/aws",
+            config=VaultAutoAuthAWS(role="vector-log-proxy"),
+        ),
+        sink=[VaultAutoAuthSink(type="file", config=[VaultAutoAuthFileSink()])],
+    ),
+    template=[
+        VaultTemplate(
+            contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+            "{{ printf .Data.key }}{{ end }}",
+            destination=Path(f"{vector_config.tls_config_directory}/odl_wildcard.key"),
+        ),
+        VaultTemplate(
+            contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+            "{{ printf .Data.value }}{{ end }}",
+            destination=Path(f"{vector_config.tls_config_directory}/odl_wildcard.cert"),
+        ),
+    ],
+)
+vault = Vault(
+    version=VERSIONS["vault"],
+    configuration={Path("vault.json"): vault_config},
+)
+consul = Consul(version=VERSIONS["consul"], configuration=consul_configuration)
+consul_template_config = ConsulTemplate()
+
+hashicorp_products = [vault, consul, consul_template_config]
+install_hashicorp_products(hashicorp_products)
+
+
+# # Install vector
+# vector_config.configuration_templates[
+#     TEMPLATES_DIRECTORY.joinpath("vector", "vector-log-proxy.yaml.j2")
+# ] = {}
+
+install_vector(vector_config)
+vault_template_permissions(vault_config)
+configure_vector(vector_config)
+vector_service(vector_config)
+
+for product in hashicorp_products:
+    configure_hashicorp_product(product)
+
+
+deploy_docker()
+deploy_docker_compose()
+
+if host.get_fact(HasSystemd):
+    register_services(hashicorp_products, start_services_immediately=False)
+    proxy_consul_dns()
+    watched_vector_files = [
+        f"{vector_config.tls_config_directory}/odl_wildcard.cert",
+        f"{vector_config.tls_config_directory}/odl_wildcard.key",
+    ]
+    service_configuration_watches(
+        service_name="vector", watched_files=watched_vector_files
+    )
```

Comment:
We don't need these lines because they're for setting up the TLS certificates that are used in the Vector listener for HTTP inputs which we won't be using for this instance.